subsys/logging/CMakeLists.txt

```diff
@@ -4,4 +4,8 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-zephyr_sources_ifdef(CONFIG_LOG_RPC log_rpc.c)
+if (CONFIG_LOG_FORWARDER_RPC OR CONFIG_LOG_BACKEND_RPC)
```

Comment:
Does this condition have a sense in general? Is it possible to have logging without Zephyr?

Reply:
I don't understand the question, I just didn't want to create a Zephyr library if neither log forwarder nor backend are enabled.

Reply:
Question was mostly about ability to run server's part without Zephyr considering that Bluetooth Host is part of Zephyr.

Reply:
Sorry, but I still don't get it :) This library depends on Zephyr logging subsystem but the condition is needed not because I think it's possible to have logging without Zephyr so I need more context.

Reply:
Considering said above means it logging was enabled, then Zephyr is present. Am I wrong?

Reply:
No, you're correct. How is it related to the condition though?

Reply:
This is just clean up. The change isn't worth time that we've already for discussion. If you feel that you are 100% right in your assurance, just leave at `as is`.

Reply:
It seems you assume that this code intents to do something which in fact it does not. Hence I'm confused :)
If I remove the `if`, yes - it will work, but it will generate a warning when neither log forwarder nor backend are enabled. It is because `zephyr_library()` will just call CMake's `add_library()` under the hood but then no `target_sources()` will be called for the library.

To that end, Zephyr provides `zephyr_library_property(ALLOW_EMPTY TRUE)` functionality, which I can use instead of the condition if you prefer it, but I saw such a condition in other places in NCS as well.